python/example.py

```diff
@@ -0,0 +1,91 @@
+from pyarrow import flight
+import argparse
+import traceback
+
+# DremioServerClientAuthHandler extends flight.ClientAuthHandler
+class DremioBasicServerAuthHandler(flight.ClientAuthHandler):
+  # Constructor
+  def __init__(self, username, password):
+    self.username = username
+    self.password = password
+    super(flight.ClientAuthHandler, self).__init__()
+
+  def authenticate(self, outgoing, incoming):
+    basicAuth = flight.BasicAuth(self.username, self.password)
+    outgoing.write(basicAuth.serialize())
+    self.token = incoming.read()
+
+  def get_token(self):
+    return self.token
+
+def parseArguments():
+  """
+  Parses the command-line arguments supplied to the script.
+  """
+  # Process command line arguments.
+  parser = argparse.ArgumentParser()
+  parser.add_argument('-host', '--hostname', type=str, help='Dremio co-ordinator hostname', default='localhost')
+  parser.add_argument('-port', '--flightPort', type=str, help='Dremio flight server port', default='32010')
+  parser.add_argument('-user', '--username', type=str, help='Dremio username', required=True)
+  parser.add_argument('-pass', '--password', type=str, help='Dremio password', required=True)
+  parser.add_argument('-query', '--sqlquery', type=str, help='SQL query to test', required=False)
+  parser.add_argument('-tls', '--tls', dest='tls', help='Enable encrypted connection', required=False, default=False, action='store_true')
+  parser.add_argument('-certs', '--trustedCertificates', type=str, help='Path to trusted certificates for encrypted connection', required=False)
+  return parser.parse_args()
+
+def connectToFlightServer(hostname, flightPort, username, password, sqlquery, tls, certs):
+  """
+   Connects to the Dremio Server and executes the specified Flight API calls.
+
+  Args:
+      hostname : Hostname to connect to.
+      port : Flight server port.
+      username : Dremio username to use for authentication.
+      password : Dremio password to use for authentication.
+      sqlquery : SQL query to test.
+      tls : Enable encrypted connection.
+      certs : Path to trusted certificates for a TLS connection.
+  """
+
+  try:
+    scheme = "grpc+tcp"
```

Comment:
What is this? If this is meant as a true example, it's missing a fair amount of comments explaining what is happening.

Reply:
Added comments throughout the application.